src/zino/trapobservers/link_traps.py

```diff
@@ -0,0 +1,192 @@
+"""This module implements link trap handling"""
+
+import logging
+import re
+from datetime import datetime, timedelta
+from typing import Optional, Tuple
+
+import zino.time
+from zino.statemodels import DeviceState, InterfaceState, Port, PortStateEvent
+from zino.tasks.linkstatetask import LinkStateTask
+from zino.trapd import TrapMessage, TrapObserver
+
+_logger = logging.getLogger(__name__)
+TRAP_WINDOW = timedelta(minutes=5)
+
+
+class LinkTrapObserver(TrapObserver):
+    WANTED_TRAPS = {
+        ("IF-MIB", "linkUp"),
+        ("IF-MIB", "linkDown"),
+    }
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._last_same_trap: dict[Tuple[str, int], datetime] = {}
+
+    def handle_trap(self, trap: TrapMessage) -> Optional[bool]:
+        _logger.debug("%s: %s (vars: %s)", trap.agent.device.name, trap.name, ", ".join(trap.variables))
+
+        if "ifIndex" in trap.variables:
+            ifindex = trap.variables.get("ifIndex").value
+        else:
+            _logger.warning("%s: %s trap contained no ifIndex value, ignoring", trap.agent.device.name, trap.name)
+            return False
```

Comment:
`"ifIndex"`is used a few times, so maybe make this a constant?

Could do the same with `"linkUp"` and  `"linkDown"` for consistencies sake